app/services/twilio_api_service.rb

```diff
@@ -86,28 +87,16 @@ def send_whatsapp(recipient_number:, message:, callback_url: nil, context: {})
   private
 
   def send_twilio_message(sender_number, recipient_number, message, callback_url, context)
+    Sentry.set_tags(context)
     client.messages.create(
       from: sender_number,
       to: recipient_number,
       status_callback: callback_url,
       body: message
     )
+    metrics.increment("#{communication_type}.sent")
   rescue Twilio::REST::TwilioError => exception
-    if exception.respond_to?(:code)
-      log_error(exception, sender_number, recipient_number, context)
-      nil
-    else
-      raise Error.new("Error while calling Twilio API", exception_message: exception.to_s, context: context)
-    end
-  end
-
-  def log_error(e, sender_number, recipient_number, context)
-    logger.info({
-      class: self.class.name,
-      error: e,
-      sender: sender_number,
-      recipient: recipient_number,
-      **context
-    })
+    metrics.increment("#{communication_type}.errors")
+    raise Error.new("Error while calling Twilio API: #{exception.message}")
```

Comment:
We don't need to add more state to the custom error ...the original error will be capture implicitly by ruby (and then we will see it in sentry).

Reply:
Interesting. I didn't know about this Sentry.set_tags.